### PR TITLE
fix(synthetic-shadow): remove patch flags

### LINKED ISSUE #1222: Remove the shadow dom bypass flag isGlobalPatchingSkipped
### Description
the document-shadow and document-body-shadow polyfills enforce shadow dom semantics on document and document.body. It also has a kill switch to bypass shadow dom semantics. This was added as a temporary measure and must be removed.

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

These flags are [unused](https://salesforce.quip.com/ntx3AQmbAatV) and we are in the process of [hard-blocking them in lwc-platform](https://git.soma.salesforce.com/lwc/lwc-platform/pull/997).

AIUI, these were originally experiments that we decided not to ship to avoid breaking changes.

Closes #1222.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ⚠️ Yes, it does include an observable change.

Yes, but nobody in core is using these flags.

### CODE REVIEW

File: packages/@lwc/synthetic-shadow/src/faux-shadow/html-element.ts [hunk 1]
```diff
@@ -4,12 +4,8 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import featureFlags from '@lwc/features';
 import { isNull, isFalse, defineProperties, defineProperty } from '@lwc/shared';
 
-import { getInnerText } from '../3rdparty/inner-text';
```
Comment (jodarove) [inline packages/@lwc/synthetic-shadow/src/faux-shadow/html-element.ts:10]: should we remove `3rdparty/inner-text.ts`?

File: packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts [hunk 2]
```diff
@@ -379,18 +347,12 @@ defineProperties(Element.prototype, {
...
-                    this,
-                    nodeList,
-                    ShadowDomSemantic.Disabled
-                );
-                return createStaticNodeList(filteredResults);
-            }
-
-            return createStaticNodeList(
-                getFilteredArrayOfNodes(this, nodeList, ShadowDomSemantic.Enabled)
+            const filteredResults = getFilteredArrayOfNodes(
```
Comment (jodarove) [inline packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts:350]: now that this function is never called with `ShadowDomSemantic.Enabled` as 3rd argument, should we remove that argument and the code that now is unreachable?

**Review by jodarove — CHANGES_REQUESTED** [review]

File: packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts (hunk 2 above)
Comment (nolanlawson) [reply packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts]: Good catch! Removed the enum entirely.

File: packages/@lwc/synthetic-shadow/src/faux-shadow/html-element.ts (hunk 1 above)
Comment (nolanlawson) [reply packages/@lwc/synthetic-shadow/src/faux-shadow/html-element.ts:10]: Missed that! Removed.

**Review by pmdartus — COMMENTED** [review]
Thanks for taking care of removing the unused feature flags.

Some of those flags were introduced to better align the synthetic shadow DOM impl with the shadow DOM spec. I think it important to add a comment whenever the synthetic shadow DOM impl deviates from the spec explaining that we tried to fix this behavior but were not able to enable it in production.

I also see that you got rid of certain tests, IMO we should update them with the new behavior (without the flag enabled) to avoid future regressions.

**Comment by nolanlawson** [comment]
@pmdartus Good idea. Done!

**Comment by nolanlawson** [comment]
Unfortunately it turns out that `innerText` and `outerText` _heavily_ differ between browsers. So these tests may be somewhat unstable (e.g. after a browser update).

Still, I've added the tests.

**Review by pmdartus — APPROVED** [review]
> Still, I've added the tests.

Let's see how far we can go with this approach. I don't expect those APIs to change often. If those tests break, let's delete them.

**Pull request merged**